lib/atomic_cache/atomic_cache_client.rb

```diff
@@ -162,6 +159,8 @@ def wait_for_new_value(key, options, tags)
         backoff_duration_ms = option(:backoff_duration_ms, options, backoff_duration_ms)
         sleep((backoff_duration_ms.to_f / 1000) * attempt)
 
+        # re-fetch the key each time, to make sure we're actually getting the latest key with the correct LMT
+        key = @timestamp_manager.current_key(keyspace)
```

Comment:
This line here is effectively the bug fix. Each attempt it will make sure to get the most current key.

How this is actually working in a cold start scenario is that because LMT is empty, the initial key is just `prefix.foobar`. The value that actually gets written by the generating process however is `prefix.foobar.timestamp`. The old code would continue to try and read `prefix.foobar` then exhaust it's attempts. This fix causes it to always grab the most current key, which will cause it to see `prefix.foobar.timestamp` as written by another generating process, and use that key for a value.